src/library/dao/autodjcratesdao.cpp

```diff
@@ -881,3 +881,71 @@ void AutoDJCratesDAO::slotPlayerInfoTrackUnloaded(QString group,
         }
     }
 }
+
+int AutoDJCratesDAO::getRandomTrackIdFromLibrary(void) {
+
+    // getRandomTrackId() would have already created
+    // the temporary auto-DJ-crates database.
+    QSqlQuery oQuery(m_rDatabase);
+
+    int iTrackId = -1;
+    // Get total as well as unplayed tracks
+    int iUnplayedTracks = 0, iTotalTracks = 0;
+    oQuery.prepare("SELECT COUNT(*) AS count FROM library "
+          "WHERE timesplayed = 0 "
+          "UNION ALL SELECT COUNT(*) AS count FROM "
+          "library ");
+    if (oQuery.exec()) {
+        if (oQuery.next()) {
+            iUnplayedTracks = oQuery.value(0).toInt();
+            if (oQuery.next())
+                iTotalTracks = oQuery.value(0).toInt();
+        }
+    } else {
+        LOG_FAILED_QUERY(oQuery);
+        return -1;
+    }
+
+    qDebug() << "total "<< iTotalTracks<< "unplayed " << iUnplayedTracks;
+
+    if ( iUnplayedTracks != 0) {
+        // Get an unplayed track id
+        oQuery.prepare("SELECT library.id  "
+                "FROM library "
+                "WHERE timesplayed = 0 "
+                "ORDER BY RANDOM() "
```

Comment:
@daschuer : I noticed a problem here , please tell me if I am wrong . This affects my existing code as well as the query you suggested. In case I change the query to
```
oQuery.prepare("SELECT library.id "
               "FROM library "
               "ORDER BY timesplayed "
```
the addRandom button wont function as one intends it to for the following cases. 
* Say you have only 1 unplayed track , each call to this function will retrieve the same track .
* Say you have n unplayed tracks and n+1 queries , again the result wont be random ?

Im I wrong here ? 
If not I think I should just retrieve a random track from the library . Or at the least ensure that I retrieve an unplayed track given that there are atleast MinUnplayedTracks no of unplayed tracks ?